logbook/logbook.go

```diff
@@ -207,6 +207,52 @@ func (book *Book) DeleteAuthor() error {
 	return fmt.Errorf("not finished")
 }
 
+// ResolveRef finds the identifier & head path for a dataset reference
+// implements resolve.NameResolver interface
+func (book *Book) ResolveRef(ctx context.Context, ref *dsref.Ref) (string, error) {
+	if book == nil {
+		return "", dsref.ErrNotFound
+	}
+
+	log, err := book.DatasetRef(ctx, *ref)
+	if err != nil {
+		return "", dsref.ErrNotFound
+	}
+	ref.InitID = log.ID()
+
+	if ref.Path == "" {
+		br, err := book.BranchRef(ctx, *ref)
```

Comment:
From https://github.com/qri-io/qri/blob/master/logbook/logbook.go#L708
"TODO(dustmop): Do not add new callers to this, transition away (preferring branchLog instead)"

Using branchLog will give you a strongly typed log, which can be passed to latestSavePath to make sure it's never called with a datasetLog instead.

Same with the call to `DatasetRef` above. That should use `datasetLog`, which can then use `dlog.InitID()` to get the initID.